compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java

```diff
@@ -380,6 +382,17 @@ public void visit(BLangInvocation invocationNode) {
         // eg: string modifiedStr = sampleStr.replace("hello", "Hello").<cursor>toLower();
     }
 
+    @Override
+    public void visit(BLangErrorConstructorExpr errorConstructorExpr) {
+        for (BLangExpression positionalArg : errorConstructorExpr.positionalArgs) {
+            this.acceptNode(positionalArg, this.symbolEnv);
```

Comment:
Other visit implementations seem to set the scope before calling `acceptNode`, do we need to do the same here?

@nadeeshaan

Reply:
Doesn't it handle within the expression itself? I wonder whether the logic here is to just route to the expression visitor and handle the scoping there?